From 12569170035f50496fb99f0aa1cfb9f9e9d63650 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 1 Mar 2016 12:39:02 +0100
Subject: [PATCH 0110/1345] Revert "fix: spi: a38x: Fix the value of
 size-cells parameter in DT"

commit  737a1495f8fe9ed41ecfa3782720cac52dcbaec0 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

If the cells size is set to 0, the partition describe in dts are not
created due to error "spi0.0: error parsing ofpart partition"

Reverts 'commit b0531c49d702 ("fix: spi: a38x: Fix the value of
size-cells parameter in DT")'

and also introduce relevant change for armada-382-db.dts and
armada-388-clearfog.dts.

Change-Id: Ic92073e182535a5a6ac2c584aa41e086f4fddce0
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27977
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-380-gp.dts       |    2 +-
 arch/arm/boot/dts/armada-382-db.dts       |    2 +-
 arch/arm/boot/dts/armada-385-db-ap.dts    |    2 +-
 arch/arm/boot/dts/armada-388-clearfog.dts |   20 ++++++++++++++++++++
 arch/arm/boot/dts/armada-388-db.dts       |    2 +-
 arch/arm/boot/dts/armada-388-gp.dts       |    2 +-
 arch/arm/boot/dts/armada-388-rd.dts       |    2 +-
 7 files changed, 26 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/armada-380-gp.dts b/arch/arm/boot/dts/armada-380-gp.dts
index f74db5f..b630144 100644
--- a/arch/arm/boot/dts/armada-380-gp.dts
+++ b/arch/arm/boot/dts/armada-380-gp.dts
@@ -69,7 +69,7 @@
 
 				spi-flash@0 {
 					#address-cells = <1>;
-					#size-cells = <0>;
+					#size-cells = <1>;
 					compatible = "st,m25p128";
 					reg = <0>; /* Chip select 0 */
 					spi-max-frequency = <50000000>;
diff --git a/arch/arm/boot/dts/armada-382-db.dts b/arch/arm/boot/dts/armada-382-db.dts
index 24e5dc0..bcf2624 100644
--- a/arch/arm/boot/dts/armada-382-db.dts
+++ b/arch/arm/boot/dts/armada-382-db.dts
@@ -87,7 +87,7 @@
 
 				spi-flash@0 {
 					#address-cells = <1>;
-					#size-cells = <0>;
+					#size-cells = <1>;
 					compatible = "w25q32", "jedec,spi-nor";
 					reg = <0>; /* Chip select 0 */
 					spi-max-frequency = <108000000>;
diff --git a/arch/arm/boot/dts/armada-385-db-ap.dts b/arch/arm/boot/dts/armada-385-db-ap.dts
index 291da24..80c325d 100644
--- a/arch/arm/boot/dts/armada-385-db-ap.dts
+++ b/arch/arm/boot/dts/armada-385-db-ap.dts
@@ -266,7 +266,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <54000000>;
diff --git a/arch/arm/boot/dts/armada-388-clearfog.dts b/arch/arm/boot/dts/armada-388-clearfog.dts
index 0d5f1f0..c3bfc80 100644
--- a/arch/arm/boot/dts/armada-388-clearfog.dts
+++ b/arch/arm/boot/dts/armada-388-clearfog.dts
@@ -276,4 +276,24 @@
 	 * CS2: mikrobus
 	 */
 	pinctrl-0 = <&spi1_pins &clearfog_spi1_cs_pins &mikro_spi_pins>;
+
+	spi-flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "w25q32", "jedec,spi-nor";
+		reg = <0>; /* Chip select 0 */
+		spi-max-frequency = <3000000>;
+		status = "disabled";
+	};
+
+	slic@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "mv_slic";
+		reg = <1>; /* Chip select 1 */
+		spi-max-frequency = <3000000>;
+		spi-1byte-cs;
+		spi-cpol;
+		spi-cpha;
+	};
 };
diff --git a/arch/arm/boot/dts/armada-388-db.dts b/arch/arm/boot/dts/armada-388-db.dts
index 013f4dc..a727131 100644
--- a/arch/arm/boot/dts/armada-388-db.dts
+++ b/arch/arm/boot/dts/armada-388-db.dts
@@ -270,7 +270,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
 		compatible = "w25q32", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <108000000>;
diff --git a/arch/arm/boot/dts/armada-388-gp.dts b/arch/arm/boot/dts/armada-388-gp.dts
index 228834b..158c558 100644
--- a/arch/arm/boot/dts/armada-388-gp.dts
+++ b/arch/arm/boot/dts/armada-388-gp.dts
@@ -443,7 +443,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <50000000>;
diff --git a/arch/arm/boot/dts/armada-388-rd.dts b/arch/arm/boot/dts/armada-388-rd.dts
index ddb1649..af82f27 100644
--- a/arch/arm/boot/dts/armada-388-rd.dts
+++ b/arch/arm/boot/dts/armada-388-rd.dts
@@ -136,7 +136,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <108000000>;
-- 
1.7.9.5

